src/main.rs

```diff
@@ -1,40 +1,43 @@
 use anyhow::Result;
-use clap::Parser;
-use fuelup::commands::install;
+use fuelup::download::component;
+use fuelup::{fuelup_cli, proxy_cli};
+use std::panic;
+use std::path::PathBuf;
 
-use fuelup::commands::fuelup::{self_update, FuelupCommand};
-use fuelup::commands::install::InstallCommand;
-
-#[derive(Debug, Parser)]
-#[clap(name = "fuelup", about = "Fuel Toolchain Manager", version)]
-struct Cli {
-    #[clap(subcommand)]
-    command: Commands,
+fn is_supported_component(component: &str) -> bool {
+    ["forc", "fuel-core", "forc-fmt", "forc-lsp", "forc-explore"].contains(&component)
 }
 
-#[derive(Debug, Parser)]
-enum Commands {
-    /// Installs the latest Fuel toolchain.
-    Install(InstallCommand),
-    /// Manage your fuelup installation.
-    #[clap(name = "self", subcommand)]
-    Fuelup(FuelupCommand),
+fn run() -> Result<()> {
+    let arg0 = std::env::args().next().map(PathBuf::from);
+
+    let process_name = arg0
+        .as_ref()
+        .and_then(|a| a.file_stem())
+        .and_then(std::ffi::OsStr::to_str)
+        .map(String::from);
+
+    match process_name.as_deref() {
+        Some(component::FUELUP) => fuelup_cli::fuelup_cli()?,
+        Some(n) => {
+            if is_supported_component(n) {
+                proxy_cli::proxy_run(n)?;
+            }
+        }
```

Comment:
> I guess this shouldn't happen as this main function should only get called from fuelup or a proxied component,

Yeah this was my thinking when deciding to not handle it. But you have a point since we can never be too safe

Reply:
I believe the updated code still doesn't give an error for the case where `!component::SUPPORTED_COMPONENTS.contains(&n)`. Is there any reason you didn't go with my suggestion above? Updated for the new const it might be:

```rust
Some(n) if SUPPORTED_COMPONENTS.contains(&n) => proxy_cli::proxy_run(n)?,
Some(n) => bail!("fuelup invoked with unexpected command or component {:?}", n),
```
This also ensures `proxy_run`'s original error is returned too, and isn't obscured by the new "fuelup invoked with unexpected command or component" error.

Reply:
Ahh I see what you mean now, my bad. I misunderstood your original comment and thought that the `None` match arm handled this scenario 🤦

Reply:
Should be fixed in [e4d3fc3](https://github.com/FuelLabs/fuelup/pull/73/commits/e4d3fc3910fc6bbd5b2d9e2d774e86b15c97230e) - also printing the error using `error!`.